eng/Versions.props

```diff
@@ -139,6 +139,8 @@
     <!-- Packages from dotnet/arcade -->
     <MicrosoftDotNetBuildTasksInstallersVersion>7.0.0-beta.22367.1</MicrosoftDotNetBuildTasksInstallersVersion>
     <MicrosoftDotNetBuildTasksTemplatingVersion>7.0.0-beta.22367.1</MicrosoftDotNetBuildTasksTemplatingVersion>
+    <!-- Packages from dotnet/source-build-externals -->
+    <MicrosoftSourceBuildIntermediatesourcebuildexternalsVersion>7.0.0-alpha.1.22369.2</MicrosoftSourceBuildIntermediatesourcebuildexternalsVersion>
```

Comment:
I made this change in reaction to some code checks that that happen in https://github.com/dotnet/aspnetcore/blob/0602a3266ea9d116df479d558cd99e854d4bc03d/eng/scripts/CodeCheck.ps1.

@dougbu or @wtgodbe might know if there is a way to exempt a dependency from having a version required in the property file even if it is in the dependency.

Reply:
That's fine.  I just wanted to call it out since these are no references to the property.

Reply:
Why does the dependency need to appear in Version.Details.xml❔

Reply:
source-build also gets metadata from version.details in order to construct its graph, it has to be there to resolve the prebuilt. That probably means we should update https://github.com/dotnet/aspnetcore/blob/0602a3266ea9d116df479d558cd99e854d4bc03d/eng/scripts/CodeCheck.ps1#L83-L86 since the assumption is no longer valid that version.details & versions.props need to match.

Reply:
@wtgodbe is correct.  In addition, the repo's source-build leg restores the packages from any of these dependencies marked with the source-build metadata so that the source-built version of those dependencies are used.  This is how we mimic the tarball build at the repo level.  In context of this PR, this is where the source-build version of `System.IdentityModel.Tokens.Jwt` comes from.

Reply:
Alright, I prefer we leave CodeCheck.ps1 alone and just add the property here

Reply:
I think that's fine for now, but long term I think we should change CodeCheck to ignore mismatches when the `SourceBuild` metadata is present. If you agree I can open an issue

Reply:
I don't agree it's worthwhile adding the complication. Easy enough to add a property